build.gradle

```
@@ -72,11 +77,22 @@ allprojects {
         excludeFilter = file("${project.rootDir}/spotbugs-exclude.xml")
     }
 
-    tasks.withType(com.github.spotbugs.SpotBugsTask) {
+    tasks.withType(SpotBugsTask) {
         reports.xml.enabled = false
         reports.html.enabled = true
     }
 
+    codeCheckTasks.dependsOn(tasks.withType(Checkstyle))
+    codeCheckTasks.dependsOn(tasks.withType(RatTask))
+    codeCheckTasks.dependsOn(tasks.withType(SpotBugsTask))
+
+    tasks.withType(Test) {
+        shouldRunAfter(codeCheckTasks)
+        shouldRunAfter(tasks.withType(Checkstyle))
+        shouldRunAfter(tasks.withType(RatTask))
+        shouldRunAfter(tasks.withType(SpotBugsTask))
+    }
+
```

So, you want to achieve the effect of `check` with `test`? If you run `./gradlew check`, it should run test and the checks specified.

---

I think it is described in line 59:
> // Force checkstyle, rat, and spotBugs to run before test tasks for faster feedback

Just tried running `./gradlew test` locally (by removing license header from a couple of java files), and it doesn't modify the test behavior. 

`./gradlew check` fails fast before running tests.